### DISCUSSION TITLE
How to prevent convert from overwriting files?

### DISCUSSION THREAD

**Discussion by feklee** [ImageMagick/ImageMagick#2654] [Help] [answered]
Example command line:

    convert test.pdf %d.jpg

I wonder if there is a way to prevent `convert` from overwriting existing files. Of course, I could check if the files exist before, but as a sequence of images is created that'd be a bit painful.

**Comment by snibgo** [comment, marked as answer]
IM has no mechanism to prevent it from overwriting files.

Your shell has commands to make files read-only. For example, in Windows, "attrib +R x.png". Then an IM command that attempts to overwrite the file will fail.

**Comment by feklee** [comment]
> Your shell has commands to make files read-only.

Thanks, but that's not a solution. The idea is to prevent a script built around ImageMagick to change any existing files on disk.

I was hoping that there is something like the `--interactive` option in Gnu `rm`.

**Comment by snibgo** [comment]
The IM command-line tools have no interactive facilities.

**Comment by feklee** [comment]
Interactive is not even necessary. As a safety feature, it would be nice to just let IM abort with return code ≠ 0 and do nothing.

**Comment by fmw42** [comment]
Test if file exists, is readable and not empty, then do not process and output message; otherwise do convert processing

```
infile="lena.png"
[ -f $infile -a -r $infile -a -s $infile ] && echo "file exists" || convert "$infile" <some processing> "$infile"
```

file exists

**Reply by feklee** [reply]
That's the solution for a different problem.

**Comment by hakanai** [comment]
I found that `convert` refused to overwrite the output file if it was already there, and had to `rm -f` it myself before running it. Worst thing is, the exit code was still 0.

**Comment by snibgo** [comment]
`convert` is deprecated. Use `magick` instead.

I have never seen `convert` or `magick` fail to overwrite an existing file, unless the file is read-only.

Please show your command, and "`ls -l`" or equivalent for your file.
